src/main/java/org/cryptomator/windows/autostart/WindowsAutoStart.java

```diff
@@ -59,15 +69,24 @@ public synchronized void enable() throws ToggleAutoStartFailedException {
 	@Override
 	public synchronized void disable() throws ToggleAutoStartFailedException {
 		try {
-			Files.delete(absoluteStartupEntryPath);
-			LOG.debug("Successfully deleted {}.", absoluteStartupEntryPath);
+			Files.delete(absStartupEntryPathSupplier.get());
+			LOG.debug("Successfully deleted {}.", absStartupEntryPathSupplier.get());
+		} catch (NoSuchElementException e) {
+			throw new ToggleAutoStartFailedException("Disabling auto start failed using startup folder: Name of shell link is not defined.");
 		} catch (NoSuchFileException e) {
 			//also okay
-			LOG.debug("File {} not present. Nothing to do.", absoluteStartupEntryPath);
+			LOG.debug("File {} not present. Nothing to do.", absStartupEntryPathSupplier.get());
 		} catch (IOException e) {
 			LOG.debug("Failed to delete entry from auto start folder.", e);
 			throw new ToggleAutoStartFailedException("Disabling auto start failed using startup folder.", e);
 		}
 	}
 
+	private String getShellLinkName() throws NoSuchElementException {
```

Comment:
throwing a `NoSuchElementException` just because splitting a path doesn't work as expected is very rude.